pkg/scalers/elasticsearch_scaler.go

```diff
@@ -0,0 +1,277 @@
+	password           string
+	indexes            []string
+	searchTemplateName string
+	parameters         []string
+	valueLocation      string
+	targetValue        int
+}
+
+var elasticsearchLog = logf.Log.WithName("elasticsearch_scaler")
+
+// NewElasticsearchScaler creates a new elasticsearch scaler
+func NewElasticsearchScaler(config *ScalerConfig) (Scaler, error) {
+	meta, err := parseElasticsearchMetadata(config)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing elasticsearch metadata: %s", err)
+	}
+
+	esClient, err := newElasticsearchClient(meta)
+	if err != nil {
+		return nil, fmt.Errorf("error getting elasticsearch client: %s", err)
+	}
+	return &elasticsearchScaler{
+		metadata: meta,
+		esClient: esClient,
+	}, nil
+}
+
+const defaultUnsafeSsl = false
+
+func parseElasticsearchMetadata(config *ScalerConfig) (*elasticsearchMetadata, error) {
+	meta := elasticsearchMetadata{}
+
+	var err error
+	addresses, err := GetFromAuthOrMeta(config, "addresses")
+	if err != nil {
+		return nil, err
+	}
+	meta.addresses = splitAndTrimBySep(addresses, ",")
+
+	if val, ok := config.TriggerMetadata["unsafeSsl"]; ok {
+		meta.unsafeSsl, err = strconv.ParseBool(val)
+		if err != nil {
+			return nil, fmt.Errorf("error parsing unsafeSsL: %s", err)
+		}
+	} else {
+		meta.unsafeSsl = defaultUnsafeSsl
+	}
+
+	if val, ok := config.AuthParams["username"]; ok {
+		meta.username = val
+	} else if val, ok := config.TriggerMetadata["username"]; ok {
+		meta.username = val
+	}
+
+	if config.AuthParams["password"] != "" {
+		meta.password = config.AuthParams["password"]
+	} else if config.TriggerMetadata["passwordFromEnv"] != "" {
+		meta.password = config.ResolvedEnv[config.TriggerMetadata["passwordFromEnv"]]
+	}
+
+	index, err := GetFromAuthOrMeta(config, "index")
+	if err != nil {
+		return nil, err
+	}
+	meta.indexes = splitAndTrimBySep(index, ";")
+
+	meta.searchTemplateName, err = GetFromAuthOrMeta(config, "searchTemplateName")
+	if err != nil {
+		return nil, err
+	}
+
+	if val, ok := config.TriggerMetadata["parameters"]; ok {
+		meta.parameters = splitAndTrimBySep(val, ";")
+	}
+
+	meta.valueLocation, err = GetFromAuthOrMeta(config, "valueLocation")
+	if err != nil {
+		return nil, err
+	}
+
+	targetValue, err := GetFromAuthOrMeta(config, "targetValue")
+	if err != nil {
+		return nil, err
+	}
+	meta.targetValue, err = strconv.Atoi(targetValue)
+	if err != nil {
+		return nil, fmt.Errorf("targetValue parsing error %s", err.Error())
+	}
+
+	return &meta, nil
+}
+
+// newElasticsearchClient creates elasticsearch db connection
+func newElasticsearchClient(meta *elasticsearchMetadata) (*elasticsearch.Client, error) {
+	config := elasticsearch.Config{Addresses: meta.addresses}
+	if meta.username != "" {
+		config.Username = meta.username
+	}
+	if meta.password != "" {
+		config.Password = meta.password
+	}
+
+	if meta.unsafeSsl {
+		tr := &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		}
+		config.Transport = tr
+	}
```

Comment:
I'm not an expert in elastic and maybe it's not possible but, could we avoid the if block using something like this?
```suggestion
	config.Transport = &http.Transport{
	 	TLSClientConfig: &tls.Config{InsecureSkipVerify: meta.unsafeSsl},
	}
```
We are using that approach inside utils [package for the HTTPClient](https://github.com/kedacore/keda/blob/main/pkg/util/http.go#L40-L45)

Reply:
make senses. I dig further to check